pysat/_instrument.py

```diff
@@ -918,12 +918,22 @@ def __setitem__(self, key, new_data):
             # input dict must have data in 'data',
             # the rest of the keys are presumed to be metadata
             in_data = new.pop('data')
+
+            # TODO(#908): remove code below with removal of 2d pandas support.
             if hasattr(in_data, '__iter__'):
                 if isinstance(in_data, pds.DataFrame):
                     pass
                     # filter for elif
                 elif isinstance(next(iter(in_data), None), pds.DataFrame):
                     # Input is a list_like of frames, denoting higher order data
+                    warnings.warn(" ".join(["Support for 2D pandas instrument",
+                                            "data has been deprecated and will",
+                                            "be removed in 3.2.0+.  Please",
+                                            "update to use an xarray",
+                                            "instrument type by setting",
+                                            "`pandas_format=False`."]),
```

Comment:
I would recommend against giving partial instructions for switching, since doing this alone will not work.

Reply:
How about "Support for 2D pandas instrument data has been deprecated and will be removed in 3.2.0+. if this is a custom instrument module, please construct the `load` statement to use an `xarray.Dataset`."?

Reply:
How about "Support for 2D pandas instrument data has been deprecated and will be removed in 3.2.0+. Please either raise an issue with the developers or modify the load statement to use an xarray.Dataset."